### PR TITLE
Code signing implemented for Android

### PR DESCRIPTION
**Pull request by KamenGoranchev**
Code signing implemented for Android. As a reference implementation was taken the react native implementation. The requirement is to have public key in the `config.xml` with a key `CodePushPublicKey`. That same as the ios implementation of code signing.

### CODE REVIEW

**Comment by ruslan-bikkinin** [comment]
Thank you for your contribution to CodePush Cordova SDK @KamenGoranchev! I'll be updating this branch during this PR reviewing and will provide you explanations of what I did.

File: src/android/CodePush.java [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 import android.content.pm.PackageManager;
 import android.os.AsyncTask;
+import android.util.Base64;
+
+import com.auth0.jwt.JWTVerifier;
```
Comment (max-mironov) [inline src/android/CodePush.java]: Probably we should use anither JWT library for decoding due to current library implementation consume a lot of methods. Please refer to this issue for details: https://github.com/Microsoft/react-native-code-push/issues/1002

File: src/android/CodePush.java [hunk 2]
```diff
@@ -89,14 +102,93 @@ private boolean execVerifySignature(final CordovaArgs args, final CallbackContex
...
+                }
+
+                final PublicKey publicKey;
+                try {
+                    publicKey = parsePublicKey(stringPublicKey);
+                } catch (CodePushException e) {
+                    callbackContext.error("Error occured while creating the a public key" + e.getMessage());
+                    return null;
+                }
+
```
Comment (max-mironov) [inline src/android/CodePush.java]: nit: probably can remove redundant blank line here and below

File: src/android/CodePushException.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.microsoft.cordova;
+
+/**
+ * Created by kgoranchev on 19/10/2017.
```
Comment (max-mironov) [inline src/android/CodePushException.java]: Please remove auto generated comments, don't use it throughout the project

File: src/android/CodePush.java (hunk 1 above)
Comment (KamenGoranchev) [reply src/android/CodePush.java]: Sorry about that. I was following the react native implementation, but for some reason I was looking at old code. If it is OK for you I will try with the same library they have now:
`com.nimbusds:nimbus-jose-jwt:5.1`

File: src/android/CodePushException.java (hunk 3 above)
Comment (KamenGoranchev) [reply src/android/CodePushException.java]: Will do.

File: src/android/CodePush.java (hunk 1 above)
Comment (max-mironov) [reply src/android/CodePush.java]: That should be ok, thanks for staying in touch!

**Pull request merged**